### ISSUE TITLE
Type generation on embeddings is incorrect

### ISSUE THREAD

**Issue by mosnicholas** [supabase/postgrest-js#437]
**Describe the bug**
Embeddings type definition has been updated to `string` -- it is represented internally as a `vector`, and in code often as a `number[]` -- I think `number[]` would be a more apt definition of the type in typescript.

<img width="402" alt="Screenshot 2023-06-01 at 2 03 54 PM" src="https://github.com/supabase/cli/assets/3606778/5fd6fb97-349b-406d-83d4-32a484cd82ad">

**Comment by codesnik** [comment]
> @codesnik any thoughts?

@ChuckJonas you probably wanted to mention someone else?

**Comment by soedirgo** [comment]
@steve-chavez does PostgREST convert the JSON array into `real[]` when inserting into a `vector` column? I noticed that with plain SQL `array[1,2,3]` works, but `'[1,2,3]'::jsonb` doesn't.

**Comment by steve-chavez** [comment]
@soedirgo No. All it does is pass the json body to `json_populate_recordset`.

**Comment by soedirgo** [comment]
OK, so it seems like some numeric types can be converted to `vector`, which is why the RPC using `number[]` works. But `vector` doesn't have a cast to `json`, so the text representation is used instead [0].

As an alternative to casting, you can add create a [computed column](https://postgrest.org/en/stable/references/api/tables_views.html#computed-virtual-columns) like so:

```sql
alter table mytable add embedding_arr float4[] generated always as (embedding) stored;
```

When you query that column it'll be typed as `number[]`.

[0] from the `to_json[b]` [function description](https://www.postgresql.org/docs/current/functions-json.html#FUNCTIONS-JSON-PROCESSING-TABLE)

**Comment by bugproof** [comment]
so this doesnt work neither? `.order('attributes->>ODOValue.cast(numeric)', { ascending: true });`

**Comment by soedirgo** [comment]
@bugproof you'll need to use computed columns for that, see https://github.com/PostgREST/postgrest/issues/1651#issuecomment-729099728
